lib/endpoint.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * Endpoint
+ * @constructor
+ */
+var Endpoint = function (client) {
+	/**
+	 * Create a new endpoint for the domain
+	 * @param {String} domainId Id of domain
+	 * @param {Object} params Parameters for creating a new endpoint
+	 * @param {String} params.name The endpoint's name, which SIP clients use as the "address of record" .
+	 * @param {String} params.description String to describe the endpoint.
+	 * 0param {String} params.applicationId The id of the application associated with this endpoint.
+	 * @param {Boolean} params.enabled Allow or not to receive and make calls.
+	 * @param {Object} params.credentials Auth parameters
+	 * @param {Function} [callback] Callback with the newly created endpoint
+	 * @return {Promise} A promise for the newly created endpoint
+	 */
+	this.create = function (domainId, params, callback) {
+		params.domainId = domainId;
+		return client.makeRequest({
+			path   : "domains/" + domainId + "/endpoints",
+			method : "POST",
+			body   : params
+		})
+		.then(function (response) {
+			var enpoint = params;
+			var location = response.headers.location;
+			enpoint.id = location.substring(location.lastIndexOf("/") + 1);
+			return enpoint;
+		})
+		.asCallback(callback);
+	};
+
+	/**
+	 * Gets a list of all endpoints for the domain.
+	 * @param {String} domainId Id of domain
+	 * @param {Function} callback A callback with the list of domains
+	 * @return {Promise} A promise for the list of endpoints.
+	 */
+	this.list = function (domainId, callback) {
+		return client.makeRequest({
+			path   : "domains/" + domainId + "/endpoints",
+			method : "GET"
+		})
+		.then(function (response) {
+			return response.body;
+		})
+		.asCallback(callback);
+	};
+
```

Comment:
`GET /domains/id/endpoints/` is paginated. I updated the [documentation](http://ap.bandwidth.com/docs/rest-api/endpoints/#resourceGETv1usersuseriddomainsdomainidendpoints) to reflect that. 

Instead of exposing the `page` parameter, only document the `size`. Take a look at how we did [`applications`](https://github.com/bandwidthcom/node-bandwidth/blob/2.0-preview/lib/application.js#L20)